### PR TITLE
Add the label `sources.knative.dev/service` for the source type

### LINKED ISSUE #149: Add the eventing source label to indicate the source type
**Describe the bug**
All eventing sources should have a label to indicate the source type for all the released resources.
The similar label is `networking.knative.dev/ingress-provider` for all the serving ingresses.
I propose to add the label `contrib.eventing.knative.dev/source-provider` for all the available resources.
Since this repository is for eventing-github, we will use `contrib.eventing.knative.dev/source-provider: github`.

 **Expected behavior**
Add the label `contrib.eventing.knative.dev/source-provider: github` for all resources.

 **Knative release version**
main

 **Additional context**
Facilitate the support in knative operator

### PR DESCRIPTION
**Pull request by houshengbo**
Fixes #149

### CODE REVIEW

File: config/500-controller.yaml [hunk 1]
```diff
@@ -20,6 +20,7 @@ metadata:
   labels:
     contrib.eventing.knative.dev/release: devel
     control-plane: github-controller-manager
+    contrib.eventing.knative.dev/source-provider: github
```
Comment (matzew) [inline config/500-controller.yaml:23]: Is this something that we define in the spec somewhere?

I'd like to double check this with the Source WG lead, to make sure we have a consistent behavior for all our sources' metadata.

Can you take a look here, @n3wscott ? Thanks!

Comment (houshengbo) [reply config/500-controller.yaml:23]: @matzew @n3wscott Does it make more sense to use the label `sources.knative.dev/source-producer: github`?

File: config/500-controller.yaml [hunk 2]
```diff
@@ -20,6 +20,7 @@ metadata:
   labels:
     contrib.eventing.knative.dev/release: devel
     control-plane: github-controller-manager
+    sources.knative.dev/service: github
```
Comment (n3wscott) [inline config/500-controller.yaml:23]: I assumed you were going to put this label on the instance resource of the GitHubSource. What is the value of putting it on the controller and various resources that are used to install it? Things like mink and our other ways of combining reconcilers make this fairly low value.

**Comment by n3wscott** [comment]
it was not clear you just wanted to know where the resources came from because of installing a source. This solution is going to not work so well for me I am afraid. I need to understand why you need this.

**Comment by houshengbo** [comment]
I would like to implement the eventing sources support the same way as ingress.
First, we need to bundle the yamls under the kodata in operator. This is the structure for ingress: https://github.com/knative/operator/tree/main/cmd/operator/kodata/ingress/0.22

For ingresses, all the ingresses are put under the same directory, so they are loaded at the same time. Then, operator's reconciler uses a filter checking on the label `networking.knative.dev/ingress-provider` of each resource and picking up the one that is needed. This label is used to distinguish which ingress the resource comes from. 

For eventing sources, I am wondering if we have a similar label.

**Comment by n3wscott** [comment]
maybe @evankanderson can step in here?

**Comment by evankanderson** [comment]
> I would like to implement the eventing sources support the same way as ingress.
> First, we need to bundle the yamls under the kodata in operator. This is the structure for ingress: https://github.com/knative/operator/tree/main/cmd/operator/kodata/ingress/0.22
> 
> For ingresses, all the ingresses are put under the same directory, so they are loaded at the same time. Then, operator's reconciler uses a filter checking on the label `networking.knative.dev/ingress-provider` of each resource and picking up the one that is needed. This label is used to distinguish which ingress the resource comes from.

It's a little surprising that this uses a label on resources, given that each ingress provider releases unique files, and it seems undesirable to mix objects from different providers releases because a label was set wrong.

If you absolutely need to use labels instead of, say, file names, I'd suggest that it would make more sense to update https://github.com/knative/operator/blob/main/pkg/packages/release.go#L233 to filter the fetched manifests and add the label when the files are downloaded for the operator repo.

**Comment by evankanderson** [comment]
In particular, adding this by hand to the objects in each repo without some sort of test or automation to ensure that new object _also_ have the label seems likely to lead to a poor installation experience.

**Pull request closed without merging**